datadog-trace-utils/benches/serialization.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use criterion::{black_box, criterion_group, Criterion};
+use datadog_trace_utils::msgpack_decoder;
+use datadog_trace_utils::msgpack_encoder;
+use serde_json::{json, Value};
+
+fn generate_spans(num_spans: usize, trace_id: u64) -> Vec<Value> {
+    let mut spans = Vec::with_capacity(num_spans);
+    let root_span_id = 100_000_000_000 + (trace_id % 1_000_000);
+
+    for i in 0..num_spans {
+        // If it's the first span make it the root
+        let span_id = if i == 0 {
+            root_span_id
+        } else {
+            root_span_id + i as u64 + 1
+        };
```

Comment:
if i == 0 the `root_span_id + i as u64` is root_span_id, otherwise it's a new value?

```suggestion
        // If it's the first span make it the root
        let span_id = root_span_id + i as u64;
```